src/Nethermind/Nethermind.TxPool/TxPool.cs

```diff
@@ -550,28 +605,31 @@ public bool TryGetPendingTransaction(Keccak hash, out Transaction? transaction)
         {
             lock (_locker)
             {
-                if (!_transactions.TryGetValue(hash, out transaction))
-                {
-                    _broadcaster.TryGetPersistentTx(hash, out transaction);
-
-                    // commented out as it puts too much pressure on the database
-                    // and it not really required in any scenario
-                    // * tx recovery usually will fetch from pending
-                    // * get tx via RPC usually will fetch from block or from pending
-                    // * internal tx pool scenarios are handled directly elsewhere
-                    // transaction = _txStorage.Get(hash);
-                }
+                return _transactions.TryGetValue(hash, out transaction)
+                       || _blobTransactions.TryGetBlobTx(hash, out transaction)
+                       || _broadcaster.TryGetPersistentTx(hash, out transaction);
             }
-
-            return transaction is not null;
         }
 
+        // only for tests - to test sorting
+        internal void TryGetBlobTxSortingEquivalent(Keccak hash, out Transaction? transaction)
+            => _blobTransactions.TryGetBlobTxSortingEquivalent(hash, out transaction);
```

Comment:
maybe instead of having this methods, we should derive a TestingTxPool and expose it there?